core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -465,6 +465,17 @@ default void dropColumn(ConnectorSession session, ConnectorTableHandle tableHand
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support dropping columns");
     }
 
+    /**
+     * Drop the specified column
+     *
+     * @param field path to a field without the parent column name in a row type. DROP COLUMN a.b.c returns ['b', 'c'].
+     * Fields in ARRAY and MAP type is not yet supported.
+     */
+    default void dropField(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle column, List<String> field)
```

Comment:
add `@Experimental(...) // TODO add support for rows inside arrays and maps and for anonymous row fields`